sip/execution_control/configuration_db/redis/db_client/processing_controller_client.py

```diff
@@ -1,389 +1,257 @@
-                    self._db.delete_key(blocks)
-                else:
-                    split_key = blocks.split(':')
-                    self._db.delete_key(blocks)
-
-                    # Add a event to the processing block event list to notify
-                    # about deleting from the db
-                    self._db.push_event(self.processing_event_name, "deleted",
-                                        split_key[3])
-
-            # Add a event to the scheduling block event list to notify
-            # of a deleting a scheduling block from the db
-            self._db.push_event(self.scheduling_event_name, "deleted",
-                                block_id)
-
-    def delete_processing_block(self, processing_block_id):
-        """Delete Processing Block(s).
+        return self._db.get_list('{}:completed'.format(self._aggregate_type))
 
-        Args:
-            processing_block_id (str): Processing Block ID
+    # #########################################################################
+    # Update functions
+    # #########################################################################
 
-        Raises:
-            KeyError if the specified processing_block_id is not found.
+    def update_value(self, block_id: str, field: str, value: str):
+        """Update the value of the given block id and field.
+
+        Args:
+            block_id (str): Scheduling block instance or processing block id
+            field (str): Field of the value that will be updated
+            value(str): New value
 
         """
-        LOG.debug('Deleting Processing Block %s', processing_block_id)
-        processing_block = self._db.get_block(processing_block_id)
-
-        if not processing_block:
-            raise KeyError('Processing Block not found: {}'
-                           .format(processing_block_id))
-
-        for block in processing_block:
-            if 'processing_block' in block:
-                self._db.delete_key(block)
-
-                # Remove processing block id from scheduling block id
-                scheduling_block_id = block.split(':')
-                scheduling_block_details = self.get_block_details(
-                    [scheduling_block_id[1]])
-                for block_details in scheduling_block_details:
-                    block_list = block_details['processing_blocks']
-                    if processing_block_id in block_list:
-                        block_list.remove(processing_block_id)
-                    self.update_value(scheduling_block_id[1],
-                                      'processing_blocks', block_list)
-
-                # Add a event to the processing block event list to notify
-                # about deleting from the db
-                self._db.push_event(self.processing_event_name, "deleted",
-                                    processing_block_id)
+        self._db.set_hash_value(self.get_key(block_id, self._aggregate_type),
```

Comment:
It looks like `self.get_key()`, defined on L146, only takes 1 argument (just `block_id`).